src/storage/src/table/cell_based_table.rs

```diff
@@ -456,3 +466,66 @@ impl<S: StateStore> TableIter for CellBasedTableRowIter<S> {
         }
     }
 }
+
+/// [`CellBasedTableStreamingIter`] is used for streaming executor, and will not wait for epoch.
+pub struct CellBasedTableStreamingIter<S: StateStore> {
+    /// An iterator that returns raw bytes from storage.
+    iter: StripPrefixIterator<S::Iter>,
+    /// Cell-based row deserializer
+    cell_based_row_deserializer: CellBasedRowDeserializer,
+    pub cell_based_item: Option<(Vec<u8>, Row)>,
+}
+
+impl<S: StateStore> CellBasedTableStreamingIter<S> {
+    pub async fn new(
+        keyspace: &Keyspace<S>,
+        table_descs: &[ColumnDesc],
+        epoch: u64,
+    ) -> StorageResult<Self> {
+        let cell_based_row_deserializer = CellBasedRowDeserializer::new(table_descs.to_vec());
+        let iter = keyspace.iter(epoch).await?;
+        let iter = Self {
+            iter,
+            cell_based_row_deserializer,
+            cell_based_item: None,
+        };
+        Ok(iter)
+    }
+
+    pub async fn peek(&mut self) -> StorageResult<Option<&(Vec<u8>, Row)>> {
+        if self.cell_based_item.is_none() {
```

Comment:
This is probably the wrong behaviour. It should be wrapped in another layer of Option.

Reply:
Thanks for your comments! 
I'm working on removing this function by using async stream and this PR will be updated soon.